### PR TITLE
Allow use of generic auth

### PR DESCRIPTION
**Pull request by laurenrevere**
Allow users to pass the _requests_(http for humans) auth parameter straight through to pythosf's uses for _requests_. This way, if users don't want to use token auth, can (theoretically) use any of requests supported auth styles: http://docs.python-requests.org/en/master/user/authentication/.

Update: 
Remove all `token` arguments and add util `bearer_token_auth` to create `Auth` objects.

**Why?** 
Adds an easy way to use basic auth.

### CODE REVIEW

File: pythosf/client.py [hunk 1]
```diff
@@ -60,22 +63,22 @@ def json_api_request(self, url, method=None, item_id=None, item_type=None, attri
 
         while keep_trying:
             keep_trying = False
+            combined_headers = combine_headers(self.base_headers(token=token), headers)
```
Comment (brianjgeiger) [inline pythosf/client.py]: Is this going to save in a Jupyter notebook? I had it in the actual request to keep from storing it into a variable.

File: pythosf/client.py [hunk 2]
```diff
@@ -27,10 +28,12 @@ def base_headers(token):
         }
 
     def json_api_request(self, url, method=None, item_id=None, item_type=None, attributes=None, raw_body=None,
-                         query_parameters=None, fields=None, headers=None, retry=True, token=None):
+                         query_parameters=None, fields=None, headers=None, retry=True, token=None, auth=None):
```
Comment (brianjgeiger) [inline pythosf/client.py]: Is there an auth type that works with our token? If not, we should make one. Maybe I'll deprecate out the old token format once we're sure that works well and I can clean up a bunch of code.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Overall looks good. Make sure we're not potentially leaking data to be DRY, and probably update the README for how to do tokens properly if requests is already good for that.

File: pythosf/client.py (hunk 1 above)
Comment (laurenrevere) [reply pythosf/client.py]: Ah okay. Makes sense. Will change back! (Keep forgetting about that!)

File: pythosf/client.py (hunk 2 above)
Comment (laurenrevere) [reply pythosf/client.py]: Will check!

File: pythosf/client.py (hunk 1 above)
Comment (brianjgeiger) [reply pythosf/client.py]: Yeah, it's weird. I should just give it up, really. But I am stubborn so far.

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**